antsibull/galaxy.py

```diff
@@ -199,7 +203,8 @@ async def download(self, collection: str, version: t.Union[str, semver.Version],
                     shutil.copyfile(cached_copy, download_filename)
                 return download_filename
 
-        async with self.aio_session.get(release_url) as response:
+        async with await retry_get(self.aio_session, release_url,
```

Comment:
Interesting... so if I'm reading the code right... There's a regular asynchronous function, `retry_get()` which we await.  When it does return, it returns a context manager and then that context manager is run via async with.  Is that how it works?

Reply:
For ease of use I wish we could make `retry_get()` into a context manager itself.  (Having to type `async with await` seems liek a source of confusion in the future).  But I remember being confused when I attempted to wrap a context manager so I don't know how hard that would be.

Reply:
Yep, there needs to be a async context manager to avoid this strange syntax :) This strange syntax was easier for the moment than writing an async context manager, so I started with that. I'll try the proper solution next.

Reply:
Fixed in 3ca41bab3a11b8f033e60d80d48d182d167a47a6.